Build failure: monado

### Steps To Reproduce

Steps to reproduce the behavior:
1. build monado:
```nix
services.monado = {
  enable = true;
  defaultRuntime = true; # Register as default OpenXR runtime
};
```

### Build log
```
error: builder for '/nix/store/b9akvbi38w29xrcvlyily5didg7p4023-monado-24.0.0.drv' failed with exit code 1;
       last 10 log lines:
       > 
       > CMake Error at /nix/store/yzi080r2c1zn2jzrhcfdv7dmr92yw07l-cmake-3.29.6/share/cmake-3.29/Modules/FindCUDA.cmake:883 (message):
       >   Specify CUDA_TOOLKIT_ROOT_DIR
       > Call Stack (most recent call first):
       >   /nix/store/6f9n7734yiqy074ff5yi1r5q6910ldyp-opencv-4.9.0/lib/cmake/opencv4/OpenCVConfig.cmake:86 (find_package)
       >   /nix/store/6f9n7734yiqy074ff5yi1r5q6910ldyp-opencv-4.9.0/lib/cmake/opencv4/OpenCVConfig.cmake:108 (find_host_package)
       >   CMakeLists.txt:95 (find_package)
       >
       > 
       > -- Configuring incomplete, errors occurred!
       For full logs, run 'nix log /nix/store/b9akvbi38w29xrcvlyily5didg7p4023-monado-24.0.0.drv'.
error: 1 dependencies of derivation '/nix/store/k53wfak59vc7n5b8jkfnmr2jilfigapb-ensure-all-wrappers-paths-exist.drv' failed to build
error: 1 dependencies of derivation '/nix/store/4wcf403kdjc5n2qwrhd4fbs0gfr5iwin-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/f2v9wrri53db3d3y90bxar6j38rszlkf-nixos-system-oden-24.11.20241001.27e30d1.drv' failed to build
```

### Notify maintainers

@Scrumplex 
@expipiplus1 
@prusnak 

### Metadata

```console
[I] rei@oden ~ [1]> nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.53, NixOS, 24.11 (Vicuna), 24.11.20241001.27e30d1`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.8`
 - nixpkgs: `/nix/store/lryfc8mhk1czqsa421di2y5nzz5c3b8m-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/mo/monado/package.nix
+++ b/pkgs/by-name/mo/monado/package.nix
@@ -5,6 +5,7 @@
 , bluez
 , cjson
 , cmake
+, config
 , dbus
 , doxygen
 , eigen
@@ -51,6 +52,8 @@
 , zlib
 , zstd
 , nixosTests
+, cudaPackages
+, enableCuda ? config.cudaSupport
   # Set as 'false' to build monado without service support, i.e. allow VR
   # applications linking against libopenxr_monado.so to use OpenXR standalone
   # instead of via the monado-service program. For more information see:
@@ -84,6 +87,7 @@ stdenv.mkDerivation (finalAttrs: {
     (lib.cmakeBool "XRT_HAVE_TRACY" true)
     (lib.cmakeBool "XRT_FEATURE_TRACING" true)
     (lib.cmakeBool "XRT_HAVE_STEAM" true)
+    (lib.optionals enableCuda "-DCUDA_TOOLKIT_ROOT_DIR=${cudaPackages.cudatoolkit}")
   ];
 
   buildInputs = [

